lib/pullwr.c

```diff
@@ -137,7 +137,7 @@ static void pullwr_resize(struct pullwr *pullwr, size_t need)
 	}
 
 	niov = pullwr_iov(pullwr, iov);
-	if (niov >= 1) {
+	if (newbuf && niov >= 1) {
```

Comment:
This NULL check is superfluous. For `newbuf` to be NULL, `pullwr->valid` must be 0. When `pullwr->valid` is 0, `pullwr_iov` returns 0. Therefore `niov` will be 0 and the condition here is not met.